src/main/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervals.java

```diff
@@ -0,0 +1,122 @@
+            optional = true,
+            minValue = 0
+    )
+    private int padding = 0;
+
+    @Argument(
+            doc = "output file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    public File outputFile;
+
+
+
+    @Override
+    public void onTraversalStart() {
+        createBins();
+    }
+
+    /**
+     * Generates binning coverage in the intervals given by the user.
+     * The width of bins, the intervals and the output file's path are given by the user.
+     */
+    public void createBins() {
+        // check if the output directory exists
+        if (!outputFile.exists() && !outputFile.mkdir()) {
+            throw new RuntimeException("Unable to create file: " + outputFile.getAbsolutePath());
+        }
+
+        // check if the bin widths are set appropriately
+        if(widthOfBins <= 0) {
+            throw new IllegalArgumentException("Width of bins " + Integer.toString(widthOfBins) + " should be >= 0.");
+        }
+
+        // get the sequence dictionary
+        final SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
+        final List<SimpleInterval> intervals = hasIntervals() ? intervalArgumentCollection.getIntervals(sequenceDictionary)
+               : IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
+
+        // create an IntervalList by copying all elements of 'intervals' into it
+        IntervalList intervalList = new IntervalList(sequenceDictionary);
+        intervals.stream().map(si -> new Interval(si.getContig(), si.getStart(), si.getEnd())).forEach(intervalList::add);
+
+        // sort intervals according to their coordinates and unique them (i.e. delete duplicates)
+        intervalList.uniqued();
+
+        // pad all elements of intervalList
+        intervalList = intervalList.padded(padding,padding);
+
+        // merge those that intersect after padding
+        intervalList = IntervalList.intersection(intervalList, intervalList);
+
+        // break the intervals up to bins -- the last bin in each interval can be shorter than the others
+        IntervalList bins = new IntervalList(sequenceDictionary);
+        int bin_start, bin_end;
+        Interval new_bin;
+        for(Interval in : intervalList) {
+            bin_start = in.getStart();
+            bin_end = Math.min(bin_start + widthOfBins - 1, in.getEnd());
+            while(bin_start < in.getEnd()) {
+                new_bin = new Interval(in.getContig(), bin_start, bin_end);
+                bins.add(new_bin);
+                bin_start += widthOfBins;
+                bin_end = Math.min(bin_start + widthOfBins - 1, in.getEnd());
+            }
+        }
+
+        // write the bins into file
+        bins.write(outputFile);
+    }
+
+    @Override
+    public void traverse() { }  // no traversal for this tool!
```

Comment:
If this tool is not traversing anything, why not using a `CommandLineProgram` instead? The idea behind that simplification is:
1. Input is a sequence dictionary. My recomendation is to input a .dict file, but it could be any possible sequence dictionary source, checking for the format(.fasta with associated .dict, reads in SAM/BAM/CRAM, VCF, etc.)
2. Use and optional `IntervalArgumentCollection` to load user-specified intervals instead of all intervals.
3. Move the logic for create the bins to `doWork()`.

This will simplify the class and the command line, which will be otherwise overloaded by unused parameters provided by the `GATKTool` abstraction...

Reply:
You're right.  We are planning to expand on this tool to add the ability to take input sites (ultimately it will be `PreprocessIntervalsAndSites`), but even then perhaps a traversal is not appropriate.  You can switch to a `CommandLineProgram` for now, @MartonKN.

Reply:
@magicDGS is right that many features of GATKTool is unnecessary. However, I'm still using 'getBestAvailableSequenceDictionary()', 'hasIntervals()' and 'intervalArgumentCollection()' which are part of GATKTool.

I could, of course, copy these pieces of code from GATKTool and reuse them here, but that wouldn't be a vey good solution. Therefore, we decided with @samuelklee to keep GATKTool.

Please, let me know if you think there's a better solution.

Reply:
@MartonKN - The way to proceed if you want to get rid of the `GATKTool` unnecessary parameters is:
* Add your own argument collection for the intervals, as a `IntervalArgumentCollection`. You just need to instantiate the one that you require (either required or not). The logic for `hasInterval()` is just checking if the intervals are specified by the user (`IntervalArgumentCollection.intervalsSpecified()`), that you can access directly if the class contain the field.
* `getBestAvailableSequenceDictionary()` gets, in order of preference, the __master__, __reference__, __reads__ or __features__. The one from the __reference__ is just a dictionary file in the same folder as the reference with a concrete name (can be provided by the user as a master dictionary), but requires a reference to be provided just for that. The one for the __reads__ is coming from a merged header, and again requires the reads to be provided   just for this dictionary. The one for the features, in this case, it is not used at all... no input for features is specified in the tool. I do not know what is going to be the specific use of this tool, but from my point of view providing just a __master__ dictionary by the user is enough for the functionality; maybe a __reference__ to search for the default instead. I try to do not rely on the dictionary from the SAM/BAM files, because in principle it is not required by the specs.

@samuelklee - When you are talking about the way of processing interval and sites together, are you referring to something like an sliding-window traversal of sites (variants, reads, etc.)? If that's the case, maybe it is worthy that you have a look to my proposal of a sliding-window traversal for both reads and/or variants (on hold for the longest time in https://github.com/broadinstitute/gatk/pull/1708)

Reply:
Thanks, @magicDGS, for those detailed comments/instructions.  Given this is his first PR, we decided to keep things simple for @MartonKN and leave this as a `GATKTool` for now.

The additional processing of sites that may ultimately be done by a future version of this tool will require a very simple traversal, so your sliding-window traversal is probably more sophisticated than we'll need---looks like a worthy proposal, though!